Client/Frontend/Browser/TabLocationView.swift

```diff
@@ -227,7 +227,11 @@ class TabLocationView: UIView {
             urlTextField.attributedText = attributedString
         } else {
             // If we're unable to highlight the domain, just use the URL as is.
-            urlTextField.text = url?.absoluteString
+            if let utf8Host = url?.host?.asciiHostToUTF8() {
+                urlTextField.text = url?.absoluteString.stringByReplacingOccurrencesOfString((url?.host)!, withString: utf8Host)
```

Comment:
...then just use `stringByReplacingOccurrencesOfString(host, ...)` instead of `(url?.host)!`.